lib/newman_conway.py

```
@@ -1,10 +1,39 @@
-
-
-# Time complexity: ?
-# Space Complexity: ?
 def newman_conway(num):
     """ Returns a list of the Newman Conway numbers for the given value.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+ # P(1) = 1
+    # P(2) = 1
+    # for all n > 2
+    # P(n) = P(P(n - 1)) + P(n - P(n - 1))
+    # use memoization
+    if num == 0:
+        raise ValueError()
```

We should raise this error for _any_ value below the valid starting point of the sequence:
```py
    if num <= 0:
        raise ValueError()
```